tools/benchmark/oio-bench-account-container-update.py

```diff
@@ -0,0 +1,151 @@
+        'objects': 0,
+        'bytes': 0,
+        'damaged_objects': 0,
+        'missing_chunks': 0,
+        'mtime': time.time()
+    }
+    # Create the "main" shard
+    try:
+        API.container_update(ACCOUNT, prefix, up_req)
+        RESULTS.put(prefix)
+    except Exception as err:
+        print(err)
+
+    # Loop on all other shards
+    while True:
+        name = prefix + '%2F' + str(iteration % SHARDS)
+        mtime = time.time()
+        iteration += 1
+        # Poor man's random
+        up_req['objects'] = int(mtime) % iteration
+        up_req['bytes'] = up_req['objects'] * 42
+        up_req['mtime'] = mtime
+        try:
+            API.container_update(ACCOUNT, name, up_req)
+            RESULTS.put(name)
+            sleep(0)
+        except Exception as err:
+            print(err)
+
+
+def main(myid, queue, concurrency, delay=5.0, duration=60.0):
+    counter = 0
+    created = list()
+    now = start = checkpoint = time.time()
+    POOL.starmap(create_loop, [('buck-%d-%d' % (myid, n), )
+                               for n in range(concurrency)])
+    while now - start < duration:
+        res = RESULTS.get()
+        counter += 1
+        if now - checkpoint > delay:
+            print("Proc %d: %d updates in %fs, %f updates per second." % (
+                  myid, counter, now - checkpoint,
+                  counter / (now - checkpoint)))
+            counter = 0
+            checkpoint = now
+        created.append(res)
+        now = time.time()
+    for coro in POOL.coroutines_running:
+        coro.kill()
+    while not RESULTS.empty():
+        created.append(RESULTS.get(block=False))
+    end = time.time()
+    rate = len(created) / (end - start)
+    print("Proc %d: end. %d updates in %fs, %f updates per second." % (
+          myid, len(created), end - start, rate))
+    time.sleep(1)
+    print("Proc %d: cleaning..." % myid)
+    del_req = {'dtime': time.time()}
+    # Do not delete twice (or an exception is raised)
+    uniq_ct = set(created)
+    for _ in POOL.starmap(API.container_update,
+                          [(ACCOUNT, n, del_req) for n in uniq_ct]):
+        pass
+    POOL.waitall()
+    queue.put(rate)
+    return 0
+
+
+if __name__ == '__main__':
+    import os
+    import sys
+    if len(sys.argv) > 1:
+        if sys.argv[1] in ('-h', '-help', '--help'):
+            print(__doc__ % sys.argv[0])
+            sys.exit(1)
+        else:
+            N_PROC = int(sys.argv[1])
+    else:
+        N_PROC = 1
+    COROS = 10
+    NS = os.getenv('OIO_NS', 'OPENIO')
+    ACCOUNT = os.getenv('OIO_ACCOUNT', 'benchmark')
+    API = AccountClient({'namespace': NS}, pool_maxsize=COROS+1)
+    RESULTS = LightQueue(COROS * 10)
+    POOL = GreenPool(COROS)
+    QUEUE = multiprocessing.Queue()
+    PROCESSES = list()
+    for sub in range(N_PROC):
+        PROCESSES.append(multiprocessing.Process(
+            target=main, args=(sub, QUEUE, COROS)))
+    for proc in PROCESSES:
+        proc.start()
```

Comment:
`multiprocessing` is monkey patch with `eventlet`. Here we spawn greenlet workers, not new processes.

Reply:
The tool has been reworked. I can confirm that `top` shows several processes.